### ISSUE TITLE
the result of turf/ellipse is not exact at high lattitudes

### ISSUE THREAD

**Issue by hadbn** [Turfjs/turf#2736]
The fonction turf/ellipse has the good behavior when plotting small ellipses at low lattitudes.
However when plotting "big ellipses" close to the poles, the earth curvature is poorly taken into account, leading to problems with the resulting geometry.

Reproduced with Turf version 7.1 running at https://turf-sandbox.netlify.app/ with the following code :

```
const p = turf.point(
  [0., 60.]
);
const c = turf.polygon(turf.circle(p, 2000).geometry.coordinates, {fill: '#0FF'});
const e = turf.polygon(turf.ellipse(p, 2000, 2000).geometry.coordinates, {fill: '#ff0000'});;

return turf.featureCollection([c, e]);
```

![image](https://github.com/user-attachments/assets/230471bd-1cbb-425d-b037-c1afe0457d3a)

The two geometries should be exactly superimposed. Furthermore, taking the projection into account would mean having a “wider” ellipse on this projection near the pole than near the equator, which is not the case. This is due to the use of `rhubDestination` instead of `destination` (which uses the Harvesine formula), which leads to an incorrect calculation of the coordinates of the points of the ellipse.

This second example shows an other issue :

```
const p = turf.point(
  [0., 60.]
);
const c = turf.polygon(turf.ellipse(p, 2000, 2000, {angle: 0}).geometry.coordinates, {fill: '#0FF'});
const e = turf.polygon(turf.ellipse(p, 2000, 2000, {angle: 90}).geometry.coordinates, {fill: '#ff0000'});;

return turf.featureCollection([c, e]);
```

![image](https://github.com/user-attachments/assets/a1b4f826-9ae2-4786-8cea-f84c6e9927ae)

Once more, both geometry should be exactly superimposed. This time, the imprecision is due to the fact that the geometry is rotated once being completly calculated. This leads to bad behavior because it does not respect the projection.

**Comment by smallsaucepan** [comment]
Thanks for raising this @hadbn. Did a quick and dirty replacement using destination() instead of rhumbDestination() and got the below (blue - circle, red - 7.1.0 ellipse, yellow - my ellipse)

![Screenshot 2024-10-25 at 9 40 46 PM](https://github.com/user-attachments/assets/f462c07f-a5f1-4442-96a9-f5f01cc8d824)

Different, though still not correct. So there might be a bit more going on.

Ok, think I've got it. Below are two ellipses (1000x2000) with origins at [0, 0] and [0, 60].

![Screenshot 2024-10-25 at 9 58 20 PM](https://github.com/user-attachments/assets/9d27abef-6247-4163-88c4-6e5328d00d0f)

It looks like ellipse() generates the points as if the x and y coordinate were uniform, like on grid paper. Except x and y are actually degrees longitude and latitude, meaning as you get further North each longitude grid division gets narrower. This is why the Northern ellipse squashes toward the top.

What do you think? Would that explain it? I'm not sure though why it's bigger overall than the equatorial one.

**Comment by hadbn** [comment]
I agree with your interpretation. Changing from `rhumbDestination` to `destination` was needed yet it was not the only problem.

_I'm not sure though why it's bigger overall than the equatorial one._
-> According to me, both x and y are streched when going North or South from the equator. So two circles having the same radius will not look the same on the projection we use, when one is closer to the pole than the other.
For instance :
```
const c = turf.polygon(turf.circle([0., 0.], 2000).geometry.coordinates, {fill: '#0FF'});
const e = turf.polygon(turf.circle([0., 60.], 2000).geometry.coordinates, {fill: '#ff0000'});;

return turf.featureCollection([c, e]);
```
will result in :

![image](https://github.com/user-attachments/assets/b4930ff0-5b7f-4b25-8e25-e702b584af77)


I opened a pull request with all the changes that are -in my opinion- necessary to resolve this issue : https://github.com/Turfjs/turf/pull/2739

Do you see any other problem ?
